src/anomalib/metrics/per_image/utils.py

```diff
@@ -0,0 +1,521 @@
+
+    Args:
+        per_image_scores (Tensor): 1D Tensor of per-image scores.
+        images_classes (Tensor | None):
+            Used to filter statistics to only one class. If None, all images are considered.
+            If given, 1D Tensor of binary image classes (0 for 'normal', 1 for 'anomalous'). Defaults to None.
+        only_class (int | None):
+            Only used if `images_classes` is not None.
+            If not None, only compute statistics for images of the given class.
+            `None` means both image classes are used.
+            Defaults to None.
+        outliers_policy (str | None): How to handle outliers stats (use them?). See `OutliersPolicy`. Defaults to None.
+        repeated_policy (str | None): How to handle repeated values in boxplot stats (two stats with same value).
+                                        See `RepeatedPolicy`. Defaults to None.
+        repeated_replacement_atol (float): Absolute tolerance used to replace repeated values. Only used if
+                                            `repeated_policy` is not None (or 'none'). Defaults to 1e-2 (1%).
+
+    Returns:
+        list[dict[str, str | int | float]]: List of boxplot statistics.
+
+        Each dictionary has the following keys:
+            - 'stat_name': Name of the statistic. Possible values:
+                - 'mean': Mean of the scores.
+                - 'med': Median of the scores.
+                - 'q1': 25th percentile of the scores.
+                - 'q3': 75th percentile of the scores.
+                - 'whishi': Upper whisker value.
+                - 'whislo': Lower whisker value.
+                - 'outlo_i': low outlier value; `i` is a unique index for each low outlier.
+                - 'outhi_j': high outlier value; `j` is a unique index for each high outlier.
+            - 'stat_value': Value of the statistic (same units as `values`).
+            - 'image_idx': Index of the image in `per_image_scores` whose score is the closest to the statistic's value.
+            - 'score': The score of the image at index `image_idx` (not necessarily the same as `stat_value`).
+
+        The list is sorted by increasing `stat_value`.
+    """
+    _validate.is_tensor(per_image_scores, "per_image_scores")
+    per_image_scores_array = per_image_scores.detach().cpu().numpy()
+
+    if images_classes is not None:
+        _validate.is_tensor(images_classes, "images_classes")
+        images_classes_array = images_classes.detach().cpu().numpy()
+
+    else:
+        images_classes_array = None
+
+    # other validations happen inside `utils_numpy.per_image_scores_stats`
+
+    return utils_numpy.per_image_scores_stats(
+        per_image_scores_array,
+        images_classes_array,
+        only_class=only_class,
+        outliers_policy=outliers_policy,
+        repeated_policy=repeated_policy,
+        repeated_replacement_atol=repeated_replacement_atol,
+    )
+
+
+def compare_models_pairwise_ttest_rel(
```

Comment:
Is this used anywhere within the codebase?

Reply:
Let's say a user wants to use this, how would they use this? For example, how to get `scores_per_model: dict[str, Tensor]`?